api/src/main/java/marquez/db/OpenLineageDao.java

```diff
@@ -913,7 +913,20 @@ default DatasetRecord upsertLineageDataset(
         Optional.ofNullable(ds.getFacets())
             .map(DatasetFacets::getSchema)
             .map(SchemaDatasetFacet::getFields)
-            .orElse(null);
```

Comment:
`null` vs an _empty_ field set may have different meanings. That is, `null` may mean no schema was provided, while an empty set can be ambiguous:
* `null`, but default empty set was used
* or the schema was an empty set

In. Marquez, we do want to make that distinction here and align with OpenLineage: the schema facet is present if a schema was extracted, or `null` if not present.